src/mmw/js/src/modeling/models.js

```diff
@@ -53,48 +63,71 @@ var ProjectModel = Backbone.Model.extend({
         this.set('name', newName);
     },
 
-    saveAll: function() {
-        var self = this;
-
-        // If we haven't saved the project before,
-        // hold on to the scenarios so that we can
-        // save each one once we have an ID for the project
-        if (this.isNew) {
-            var tempScenarios = this.get('scenarios');
+    saveProjectAndScenarios: function() {
+        if (App.user.get('uid') !== this.get('uid')) {
+            return;
         }
```

Comment:
I agree to keep the check - I think the confusion here might be that there are no "buttons" that fire this off, it's really just modifications to model attributes.  The check is so that we don't try to save it, given that it will fail, not a belt and suspenders kind of check for a hidden button.